apps/qt/mainwindow.cpp

```diff
@@ -196,6 +132,71 @@ QChart* CoronanWidget::create_line_chart(
   chart->legend()->show();
 
   return chart;
+};
+
+} // namespace
+
+CoronanWidget::CoronanWidget(std::string const& api_url, QWidget* parent)
+    : QWidget{parent}, m_ui{new Ui_CoronanWidgetForm}, m_url{api_url}
+{
+  m_ui->setupUi(this);
+
+  populate_country_box();
+
+  auto country_code =
+      m_ui->countryComboBox->itemData(m_ui->countryComboBox->currentIndex())
+          .toString();
+
+  auto const country_data = get_country_data(country_code.toStdString());
+  m_chartView = new QChartView{create_line_chart(country_data)};
+  m_chartView->setRenderHint(QPainter::Antialiasing, true);
+
+  m_ui->gridLayout->addWidget(m_chartView, 2, 1);
+
+  m_ui->overviewTable->horizontalHeader()->setSectionResizeMode(
+      0, QHeaderView::ResizeToContents);
+
+  update_country_overview_table(m_ui->overviewTable, country_data);
+
+  // Set the colors from the light theme as default ones
+  auto pal = qApp->palette();
+  pal.setColor(QPalette::Window, QRgb(0xf0f0f0));
+  pal.setColor(QPalette::WindowText, QRgb(0x404044));
+  qApp->setPalette(pal);
+}
+
+CoronanWidget::~CoronanWidget() { delete m_ui; }

```

Comment:
`m_chartview` should IMO be deleted here as well.

Reply:
deleting m_chartView isnot required since it is deleted by the parent Widget.
There is however a memory leak, when replacing the Widget.